permissions/views.py

```diff
@@ -0,0 +1,206 @@
+        return redirect("permissions:home")
+
+    return render(request, "team_page.html", context)
+
+
+@require_POST
+@login_required
+def create_role(request):
+    team_id = request.POST.get("team_id")
+    role_name = request.POST.get("role_name")
+    members = request.POST.getlist("members")
+
+    try:
+        team = Team.objects.get(id=team_id)
+
+        role = Role(
+            name=role_name,
+            created_by=request.user,
+            team=team
+        )
+
+        role.save()
+
+        for member in members:
+            role.members.add(User.objects.get(username=member))
+
+        role.save()
+    except Team.DoesNotExist:
+        print("Team doesn't exist")
+        pass
+
+    return redirect('permissions:team_detail', team_id)
+
+
+@require_POST
+@login_required
+def add_permission(request):
+    team_id = request.POST.get("team_id")
+    course_id = request.POST.get("courses")
+    role_id = request.POST.get("role")
+    perm_type = request.POST.get("perm_type")
+    units = request.POST.getlist("units")
+
+    team = Team.objects.get(id=team_id)
+    course = Course.objects.get(id=course_id)
+    role = Role.objects.get(id=role_id)
+
+    # Create permission obj for each unit
+    for unit in units:
+        type, id = unit.split("_")
+
+        if type == "quiz":
+            content_object = Quiz.objects.get(pk=id)
+        else:
+            content_object = Lesson.objects.get(pk=id)
+
+        permission = Permission(
+            team=team,
+            course=course,
+            perm_type=perm_type,
+            content_object=content_object
+        )
+
+        permission.save()
+
+        permission.role.add(role)
+
+        permission.save()
+
+    return redirect('permissions:team_detail', team_id)
+
+
+def delete_permission(request, permission_id, team_id):
+    '''
+    Delete permission.
+    Only team creator can delete
+    '''
+
+    try:
+        team = Team.objects.get(pk=team_id)
+
+        if team.created_by == request.user:
+            Permission.objects.get(pk=permission_id).delete()
+
+            return redirect('permissions:team_detail', team_id)
+
+    except Team.DoesNotExist:
+        print("Team doesn't exist")
+        pass
+
+
+@login_required
+def get_modules(request):
+    ''' Get modules belonging to a course '''
+
+    course_id = request.GET.get("course_id")
+    course = get_object_or_404(Course, pk=course_id)
+    modules = course.learning_module.all()
+
+    units = []
+
+    for module in modules:
+        learning_units = module.learning_unit.all()
+        for learning_unit in learning_units:
+            if learning_unit.type == "quiz":
+                units.append({"key": "quiz_{}".format(learning_unit.quiz.id),
+                              "name": learning_unit.quiz.description})
+            else:
+                units.append(
+                    {"key": "lesson_{}".format(learning_unit.lesson.id),
+                     "name": learning_unit.lesson.name})
```

Comment:
I guess we already have a model method to get all the learning units for a module. Maybe you can use that.